### PR TITLE
CF Manual Create a Checker Chapter Update

### PR DESCRIPTION
**Pull request by jyluo**
vs RBCommons's diff, I've added some minor typo fixes and a link to chapter 29.6 (Visitor: Type rules) which states the basic typing rules applied by the BaseTypeVisitor.

### CODE REVIEW

File: checker/manual/creating-a-checker.tex [hunk 1]
```diff
@@ -14,7 +14,6 @@ \chapter{How to create a new checker\label{writing-a-checker}}
 If you only want to \emph{use} a checker, you do not need to read this
 chapter.
 
-
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: Please try to avoid gratuitous whitespace changes.  They clutter the diff and then the history.

File: checker/manual/creating-a-checker.tex [hunk 2]
```diff
@@ -37,7 +36,7 @@ \chapter{How to create a new checker\label{writing-a-checker}}
 checker.  Specific checkers that are designed for extension (besides the Subtyping
 Checker) include the Fake Enumeration Checker
 (\chapterpageref{fenum-checker}), the Units Checker
-(\chapterpageref{units-checker}), and the typestate checkers
+(\chapterpageref{units-checker}), and the Typestate Checker
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: I agree with the change to singular, but I had undone the capitalization because I see no evidence that the author uses the proper name "Typestate Checker" for his implementation".

File: checker/manual/creating-a-checker.tex [hunk 3]
```diff
@@ -109,7 +108,6 @@ \section{Relationship of the Checker Framework to other tools\label{tool-relatio
 \end{tabular}
 \end{center}
 
-
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: As above regarding whitespace.

File: checker/manual/creating-a-checker.tex [hunk 4]
```diff
@@ -153,7 +151,7 @@ \section{The parts of a checker\label{parts-of-a-checker}}
   Subtyping rules related to the type hierarchy, such as that every
   assignment and pseudo-assignment satisfies a subtyping relationship.
   Your checker automatically inherits these subtyping rules from the Base
-  Checker (Chapter~\ref{subtyping-checker}).
+  Checker See (Chapter~\ref{extending-visitor}) and (Chapter~\ref{subtyping-checker}).
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: The parentheses are off.  This is also something that I had fixed already.

File: checker/manual/creating-a-checker.tex [hunk 5]
```diff
@@ -462,7 +460,6 @@ \section{Type factory: Implicit annotations\label{writing-type-introduction}}
 
 The implicit annotations may be specified declaratively and/or procedurally.
 
-
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: It's best to have two blank lines above a section heading.  It makes the natural breaks in the document stand out even when one is reading the .tex file.

File: checker/manual/creating-a-checker.tex [hunk 6]
```diff
@@ -1271,7 +1268,6 @@ \section{Debugging options\label{debugging-options}}
 writing a checker. These are provided via the standard \code{javac} ``\code{-A}''
 switch, which is used to pass options to an annotation processor.
 
-
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: Undo this change, please.

File: checker/manual/creating-a-checker.tex [hunk 7]
```diff
@@ -37,7 +37,7 @@ \chapter{How to create a new checker\label{writing-a-checker}}
 checker.  Specific checkers that are designed for extension (besides the Subtyping
 Checker) include the Fake Enumeration Checker
 (\chapterpageref{fenum-checker}), the Units Checker
-(\chapterpageref{units-checker}), and the typestate checkers
+(\chapterpageref{units-checker}), and the typestate checker
```
Comment (mernst) [inline checker/manual/creating-a-checker.tex]: I would change "the" to "a"

**Comment by mernst** [comment]
You had added a cross-reference to an additional section of the manual.  The parenthesization was off on that edit, but I didn't object to the addition of the cross-reference.  I see that cross-reference is no longer added in this diff.  That's OK with me, but I wanted to make sure that it's intentional.

**Comment by jyluo** [comment]
The removal of the cross-reference was intentional upon further review. That paragraph begins with a cross reference to the same chapter and since it is short it seemed unnecessary to have two of the same cross-reference: "29.6 Type rules. [...] Base Checker See (Chapter 29.6) and (Chapter 22)."
